pkg/simulation/config/envoyfilter.go

```diff
@@ -0,0 +1,98 @@
+package config
+
+import (
+	structpb "github.com/golang/protobuf/ptypes/struct"
+	networkingv1alpha3 "istio.io/api/networking/v1alpha3"
+	"istio.io/client-go/pkg/apis/networking/v1alpha3"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/howardjohn/pilot-load/pkg/simulation/model"
+)
+
+type EnvoyFilterSpec struct {
+	App            string
+	Namespace      string
+	connectTimeout int
+	Parent         model.IstioAPIParent
+}
+
+type EnvoyFilter struct {
+	Spec *EnvoyFilterSpec
+}
+
+var _ model.Simulation = &EnvoyFilter{}
+
+func NewEnvoyFilter(s EnvoyFilterSpec) *EnvoyFilter {
+	return &EnvoyFilter{Spec: &s}
+}
+
+func (v *EnvoyFilter) Refresh(ctx model.Context) error {
+	v.Spec.connectTimeout = (v.Spec.connectTimeout + 1) % 10
+	return v.Run(ctx)
+}
+
+func (v *EnvoyFilter) Run(ctx model.Context) (err error) {
+	return ctx.Client.Apply(v.getEnvoyFilter())
+}
+
+func (v *EnvoyFilter) Cleanup(ctx model.Context) error {
+	return ctx.Client.Delete(v.getEnvoyFilter())
+}
+
+func (v *EnvoyFilter) getEnvoyFilter() *v1alpha3.EnvoyFilter {
+	s := v.Spec
+	spec := networkingv1alpha3.EnvoyFilter{}
+
+	name := s.Namespace
+	applyTo := networkingv1alpha3.EnvoyFilter_NETWORK_FILTER
+	context := networkingv1alpha3.EnvoyFilter_SIDECAR_OUTBOUND
+	operation := networkingv1alpha3.EnvoyFilter_Patch_INSERT_AFTER
+
+	// Apply different configurations at different levels
```

Comment:
not super clear to me what the correlation here is. I would think the level  and type is decoupled.

Reply:
The idea was to apply different configs at different levels (rootNS, NS, Application level) so that multiple EnvoyFilter configurations do conflict with each other.